zenodo/modules/sipstore/tasks.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2017 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Celery tasks for SIPStore."""
+
+from __future__ import absolute_import
+
+from celery import shared_task
+from invenio_db import db
+from invenio_sipstore.api import SIP as SIPApi
+from invenio_sipstore.archivers import BagItArchiver
+from invenio_sipstore.models import SIP
+
+
+@shared_task(ignore_result=True, rate_limit='100/m')
```

Comment:
Perhaps not, but then this task might then throttle the file checks tasks if we have a rate of >100 publishings a minute and if the total archiving task time takes longer than a minute. The low priority queue does not have any rate limiting mechanism by itself. The file check tasks are 'rate-limited' by a separate scheduler task, and here I don't have one.